5.2.cup_of_join.py

```diff
@@ -0,0 +1,18 @@
+def join(*chain, sep='-'):
```

Comment:
Dont avoid using type hints when its hard ;) read about how to do it even in these cases